src/api/src/token/coco.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2023 by Alibaba.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::token::AttestationTokenVerifier;
+use anyhow::*;
+use base64::engine::general_purpose::URL_SAFE_NO_PAD;
+use base64::Engine;
+use rsa::signature::Verifier;
+use serde_json::Value;
+
+#[derive(Default)]
+pub struct CoCoAttestationTokenVerifier {}
+
+impl AttestationTokenVerifier for CoCoAttestationTokenVerifier {
+    fn verify(&self, token: String) -> Result<String> {
+        let split_token: Vec<&str> = token.split('.').collect();
+        if !split_token.len() == 3 {
+            bail!("Illegal JWT format")
+        }
+
+        let header = URL_SAFE_NO_PAD.decode(split_token[0])?;
+        let claims = URL_SAFE_NO_PAD.decode(split_token[1])?;
+        let signature = URL_SAFE_NO_PAD.decode(split_token[2])?;
+
+        let header_value = serde_json::from_slice::<Value>(&header)?;
+        let claims_value = serde_json::from_slice::<Value>(&claims)?;
+
+        let now = time::OffsetDateTime::now_utc().unix_timestamp();
+        if let Some(exp) = claims_value["exp"].as_i64() {
+            if exp < now {
+                bail!("token expired");
+            }
+        }
+        if let Some(nbf) = claims_value["nbf"].as_i64() {
+            if nbf > now {
+                bail!("before validity");
+            }
+        }
```

Comment:
Do we always want to have expiration set on tokens?

```suggestion
        let now = time::OffsetDateTime::now_utc().unix_timestamp();
        let Some(exp) = claims_value["exp"].as_i64() else {
            bail!("token expiration unset");
        }   
        if exp < now {
             bail!("token expired");
        }
        if let Some(nbf) = claims_value["nbf"].as_i64() else {
            if nbf > now {
                bail!("before validity");
            }
        }
```